c++/include/orc/Writer.hh

```diff
@@ -0,0 +1,228 @@
+     * 0 to disable dictionary encoding.
+     * 1 to always enable dictionary encoding.
+     */
+    WriterOptions& setDictionaryKeySizeThreshold(double val);
+
+    /**
+     * Get the dictionary key size threshold.
+     */
+    double getDictionaryKeySizeThreshold() const;
+
+    /**
+     * Set Orc file version
+     */
+    WriterOptions& setFileVersion(const FileVersion& version);
+
+    /**
+     * Get Orc file version
+     */
+    FileVersion getFileVersion() const;
+
+    /**
+     * Set compression kind.
+     */
+    WriterOptions& setCompression(CompressionKind comp);
+
+    /**
+     * Get the compression kind.
+     * @return if not set, return default value which is ZLIB.
+     */
+    CompressionKind getCompression() const;
+
+    /**
+     * Set the encoding strategy.
+     */
+    WriterOptions& setEncodingStrategy(EncodingStrategy strategy);
+
+    /**
+     * Get the encoding strategy.
+     * @return if not set, return default value which is SPEED.
+     */
+    EncodingStrategy getEncodingStrategy() const;
+
+    /**
+     * Set the compression strategy.
+     */
+    WriterOptions& setCompressionStrategy(CompressionStrategy strategy);
+
+    /**
+     * Get the compression strategy.
+     * @return if not set, return default value which is speed.
+     */
+    CompressionStrategy getCompressionStrategy() const;
+
+    /**
+     * Set the padding tolerance.
+     */
+    WriterOptions& setPaddingTolerance(double tolerance);
+
+    /**
+     * Get the padding tolerance.
+     * @return if not set, return default value which is zero.
+     */
+    double getPaddingTolerance() const;
+
+    /**
+     * Set the memory pool.
+     */
+    WriterOptions& setMemoryPool(MemoryPool * memoryPool);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default memory pool.
+     */
+    MemoryPool * getMemoryPool() const;
+
+    /**
+     * Set the error stream.
+     */
+    WriterOptions& setErrorStream(std::ostream& errStream);
+
+    /**
+     * Get the error stream.
+     * @return if not set, return std::err.
+     */
+    std::ostream * getErrorStream() const;
+
+    /**
+     * Set whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     */
+    WriterOptions& setEnableStats(bool enable);
+
+    /**
+     * Get whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     * @return if not set, the default is true
+     */
+     bool getEnableStats() const;
+
+    /**
+     * Get whether or not to write row group index
+     * @return if not set, the default is false
+     */
+    bool getEnableIndex() const;
+  };
+
+  class Writer {
+  public:
+    virtual ~Writer();
+
+    /**
+     * Create a row batch for writing the columns into this file.
+     * @param size the number of rows to read
+     * @return a new ColumnVectorBatch to write into
+     */
+    virtual ORC_UNIQUE_PTR<ColumnVectorBatch> createRowBatch(uint64_t size
```

Comment:
Actually after giving a second look, I prefer keeping this method based on two reasons - 1. Reader has this method too. we probably want to be consistent. 2. Type::createRowBatch requires passing in memory pool object. In a lot of cases, we want to use the same memory pool that's used by writer. This method will ensure that.